source/CIGameScene.cpp

```diff
@@ -280,17 +299,17 @@ void GameScene::addStardust(const Size bounds) {
     
     /** Pity mechanism: The longer you haven't seen a certain color, the more likely it will be to spawn that color */
     CIColor::Value c = CIColor::getRandomColor();
-    int probSum = 0, colorCount = 6;
+    int probSum = 0;
     // Sums up the total probability space of the stardust colors, augmented by a mass correction
-    probSum = accumulate(_stardustProb, _stardustProb + colorCount, probSum) + massCorrection;
+    probSum = accumulate(_stardustProb, _stardustProb + _colorCount, probSum) + massCorrection;
     // Randomly selects a point in the probability space
-    int spawnRand = rand() % probSum;
-    for (int i = 0; i < colorCount; i++) {
+    int spawnRand = rand() % max(1, probSum);
+    for (int i = 0; i < _colorCount; i++) {
```

Comment:
you should keep the stardust spawning the same as before or update all of the stardust spawning to be consistent with _colorCount.